packages/@icon-magic/generate/src/plugins/svg-light-dark.ts

```diff
@@ -0,0 +1,172 @@
+      if (!lightFlavor) {
+        return flavor;
+      } else {
+        const darkFlavorContents = (await flavor.getContents()) as string; // .svg asset's getContents() returns a string (the svg as a string)
+        const lightFlavorContents  = (await lightFlavor.getContents()) as string;
+
+        const lightToken = params.lightToken;
+        const darkToken = params.darkToken;
+
+        const lightSvgo = new Svgo({
+          plugins: [
+            {
+              addAttributesToSVGElement: {
+                attributes: [{display: lightToken || "var(--svg-light-display)"}]
+              }
+            },
+            {
+              removeViewBox: false
+            }
+          ],
+          js2svg: { pretty: true, indent: 2 }
+        });
+
+        const darkSvgo = new Svgo({
+          plugins: [
+            {
+              addAttributesToSVGElement: {
+                attributes: [{display: darkToken || "var(--svg-dark-display)"}]
+              }
+            },
+            {
+              removeViewBox: false
+            }
+          ],
+          js2svg: { pretty: true, indent: 2 }
+        });
+
+        if (lightFlavorContents){
+          // create optimized light/dark svg assets
+          const lightAsset = await lightSvgo.optimize(lightFlavorContents); //returns a string
+          const darkAsset = await darkSvgo.optimize(darkFlavorContents);
+
+          const mixedParentSvg = await domSvgParentGenerate(lightAsset, darkAsset);
+
+          const mixedSvgo = new Svgo({
+            plugins: [
+              {
+                removeViewBox: false
+              }
+            ],
+            js2svg: { pretty: true, indent: 2 }
+          });
+
+          const mixedString = serializeToString(mixedParentSvg);
+          const mixedAsset = await mixedSvgo.optimize(mixedString); //returns a string
+
+          // write the parent svg with light/dark children svg's to the output directory
+          await fs.ensureDir(outputPath);
+          await fs.writeFile(
+            `${path.join(outputPath, imageSet)}-mixed.svg`,
+            mixedAsset.data,
+            {
+              encoding: 'utf8'
+            }
+          );
+
+          // Final new Mixed Flavor
+          const mixedFlavor: Flavor = new Flavor(icon.iconPath, {
+            name: `${imageSet}-mixed`,
+            path: `./${imageSet}-mixed.svg`,
+            colorScheme: 'mixed'
+          });
+
+          //Add new mixed flavor to icon.flavors. Is then processed during svg-generate and added to resulting iconrc.json file.
+          icon.flavors.set(
+            `${imageSet}-mixed`,
+            mixedFlavor
+          );
+          return mixedFlavor;
+        }
+      }
+    }
+    return flavor;
+  }
+};
+
+//TODO: lightAsset type would be `OptimizedSvg` from Svgo interface, but it does not accept this type, I think it is not exported.
+async function domSvgParentGenerate(lightAsset: any, darkAsset: any): Promise<SVGSVGElement> {
```

Comment:
can you add a function header comment for this like https://github.com/linkedinlabs/icon-magic/blob/c274ee1c5f003be704e4184acea2fcecb97854f0/packages/%40icon-magic/distribute/src/create-icon-template.ts#L10

Reply:
also maybe a simpler function name? like `generateMixedSvgParent` or `generateMixedSvg`. Don't think "dom" has to be in the name.

Reply:
Yes, done, Ty!

Reply:
still need to update function name